.github/workflows/suppressions.yml

```diff
@@ -0,0 +1,38 @@
+name: Suppressions
+
+on:
+  push:
+    branches:
+    - main
+    - typespec-next
+  pull_request:
+    branches:
+    - main
+    - typespec-next
+    paths:
+    - package-lock.json
+    - package.json
+    - tsconfig.json
+    - eng/tools/package.json
+    - eng/tools/tsconfig.json
+    - eng/tools/suppressions
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Use Node 20.x
+      uses: actions/setup-node@v4
+      with:
+        node-version: '20.x'
+
+    - run: npm ci
+
+    - run: npm run build
+      working-directory: ./eng/tools/suppressions
+
+    - run: npm test
```

Comment:
Basic question: I understand this file is about running unit tests in the `tools/suppressions` folder. It triggers on every commit to `main` or `typespec-next`.

Is that correct?

If someone makes a PR to `main` that would cause these tests to fail, will we notice the failure before the PR is merged, or only after?

Reply:
The action also triggers on PRs to `main` or `typespec-next` (filtered to all files that could impact the tool), so yes we should catch it.

Reply:
This file is **very** similar to `typespec-validation/ci.yml` (which is currently a devops pipeline)